File: backend/src/main/java/ch/rodano/core/services/bll/configurator/ConfiguratorConfigService.java
package ch.rodano.core.services.bll.configurator;

import java.util.List;
import java.util.UUID;

import ch.rodano.api.config.ScopeModelDTO;

public interface ConfiguratorConfigService {

	List<ScopeModelDTO> getScopeModels(UUID projectId);

	ScopeModelDTO getScopeModel(UUID projectId, UUID scopeModelId);

	ScopeModelDTO createScopeModel(UUID projectId, ScopeModelDTO scopeModel);

	ScopeModelDTO updateScopeModel(UUID projectId, UUID scopeModelId, ScopeModelDTO scopeModel);

	void deleteScopeModel(UUID projectId, UUID scopeModelId);
}

File: frontends/main/src/app/core/services/configurator-config.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {ScopeModel} from '@core/model/scope-model';
import {Observable} from 'rxjs';

@Injectable({
	providedIn: 'root'
})
export class ConfiguratorConfigService {
	constructor(private http: HttpClient) {}

	getScopeModels(projectId: string): Observable<ScopeModel[]> {
		return this.http.get<ScopeModel[]>(`/api/superuser/configurator/projects/${projectId}/config/scope-models`);
	}

	getScopeModel(projectId: string, scopeModelId: string): Observable<ScopeModel> {
		return this.http.get<ScopeModel>(`/api/superuser/configurator/projects/${projectId}/config/scope-models/${scopeModelId}`);
	}

	createScopeModel(projectId: string, scopeModel: ScopeModel): Observable<ScopeModel> {
		return this.http.post<ScopeModel>(`/api/superuser/configurator/projects/${projectId}/config/scope-models`, scopeModel);
	}

	updateScopeModel(projectId: string, scopeModelId: string, scopeModel: ScopeModel): Observable<ScopeModel> {
		return this.http.put<ScopeModel>(`/api/superuser/configurator/projects/${projectId}/config/scope-models/${scopeModelId}`, scopeModel);
	}

	deleteScopeModel(projectId: string, scopeModelId: string): Observable<void> {
		return this.http.delete<void>(`/api/superuser/configurator/projects/${projectId}/config/scope-models/${scopeModelId}`);
	}
}
